src/main/java/com/faforever/client/leaderboard/LeaderboardController.java

```diff
@@ -0,0 +1,369 @@
+        if (leagueEntry == null) {
+          selectHighestDivision();
+        } else {
+          if (leagueEntry.getSubdivision() == null) {
+            JavaFxUtil.runLater(() -> {
+              playerDivisionNameLabel.setVisible(false);
+              placementLabel.setVisible(true);
+              placementLabel.setText(i18n.get("leaderboard.placement", leagueEntry.getGamesPlayed(), season.getPlacementGames()));
+            });
+            selectHighestDivision();
+          } else {
+            Image divisionImage = assetService.loadAndCacheImage(
+                leagueEntry.getSubdivision().getImageUrl(), Path.of("divisions"), null);
+            JavaFxUtil.runLater(() -> {
+              playerDivisionNameLabel.setVisible(true);
+              placementLabel.setVisible(false);
+              playerDivisionImageView.setImage(divisionImage);
+              playerDivisionNameLabel.setText(i18n.get("leaderboard.divisionName",
+                  i18n.get(leagueEntry.getSubdivision().getDivisionI18nKey()),
+                  leagueEntry.getSubdivision().getNameKey()).toUpperCase());
+              scoreArc.setLength(-360.0 * leagueEntry.getScore() / leagueEntry.getSubdivision().getHighestScore());
+              playerScoreLabel.setText(i18n.number(leagueEntry.getScore()));
+            });
+            selectLeagueEntry(leagueEntry);
+          }
+        }
+        plotDivisionDistributions(divisions, leagueEntry);
+      }).exceptionally(throwable -> {
+        log.warn("Error while fetching leagueEntry", throwable);
+        notificationService.addImmediateErrorNotification(throwable, "leaderboard.failedToLoadEntry");
+        return null;
+      })
+    ).exceptionally(throwable -> {
+      log.warn("Error while loading division list", throwable);
+      notificationService.addImmediateErrorNotification(throwable, "leaderboard.failedToLoadDivisions");
+      return null;
+    });
+  }
+
+  private void selectHighestDivision() {
+    entryToSelect = null;
+    JavaFxUtil.runLater(() -> majorDivisionPicker.getSelectionModel().selectLast());
+  }
+
+  private void selectLeagueEntry(LeagueEntryBean leagueEntry) {
+    entryToSelect = leagueEntry;
+    // When selecting an item from majorDivisionPicker, onMajorDivisionPicked gets called automatically,
+    // but only if the selection actually changes
+    if (majorDivisionPicker.getValue() != null && correctDivisionSelected()) {
+      selectAssociatedTab();
+    } else {
+      majorDivisionPicker.getItems().stream()
+          .filter(item -> item.getDivision().getIndex() == entryToSelect.getSubdivision().getDivision().getIndex())
+          .findFirst().ifPresent(item -> JavaFxUtil.runLater(() -> majorDivisionPicker.getSelectionModel().select(item)));
+    }
+  }
+
+  private boolean correctDivisionSelected() {
+    return entryToSelect.getSubdivision().getDivision().getIndex() == majorDivisionPicker.getValue().getDivision().getIndex();
+  }
```

Comment:
No, this is a helper method to compare the division index of the entry with the selected one of the division picker. I did this, because the evaluation is very long and we need it to adjust the behaviour accordingly. For example if you searched for someone that is in silver III it will select that tab again if you switched over to gold and back to silver. However we don't want to have gold III selected. It should be the highest division when there is no specific reason to select a different tab.
I'll rename the method to make its use clearer.